### PR TITLE
Add blog

### PR DESCRIPTION
**Pull request by mashehu**

### CODE REVIEW

**Comment by ewels** [comment]
Some extra fields to show:

* Publish date
* Author(s) - GitHub handles
* Year sidebar

File: src/content/blog/2023/hello-nf-core-world.mdx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+## Greetings, nf-core Community
+
+Today marks the launch of the nf-core blog, a new avenue for keeping our community informed and connected.
+This blog will serve as a central hub for in-depth updates and discussions.
+
+## Our Vision for the Blog
+
+This platform is created with the intent to:
+
+-   Share substantive updates from our community’s activities.
```
Comment (FriederikeHanssen) [inline src/content/blog/2023/hello-nf-core-world.mdx:23]:
```suggestion
-   Share substantive updates from our community’s activities, like pipeline releases
```

Comment (FriederikeHanssen) [reply src/content/blog/2023/hello-nf-core-world.mdx:23]: or add examples elsewhere. For me it is not super clear from reading this, that it is for pipeline developers to show case their new releases :)

File: src/pages/blog/index.astro [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                                    )}
+                                    <div class="card-body">
+                                        <h2>{post.data.title}</h2>
+                                        <p class="card-text">
+                                            {async () => {
+                                                const { remarkPluginFrontmatter } = await post.render();
+                                                return remarkPluginFrontmatter.excerpt;
+                                            }}
+                                        </p>
+                                        <a href={post.slug} class="stretched-link" />
```
Comment (ewels) [inline src/pages/blog/index.astro:69]:
```suggestion
                                        <a href={'/blog/'+post.slug} class="stretched-link" />
```

**Review by ewels — COMMENTED** [review]
Getting a 404 on blog post links

File: src/content/blog/2023/hello-nf-core-world.mdx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+---
+title: 'Hello nf-core world'
+subtitle: 'The inital nf-core blog post'
+headerImage: 'https://images.unsplash.com/photo-1590005354167-6da97870c757'
+headerImageAlt: 'Green apples cut in various ways laying on an bright orange surface'
+pubDate: 2023-11-03T08:00:00.000+01:00
```
Comment (ewels) [inline src/content/blog/2023/hello-nf-core-world.mdx:6]: Need to move it into a `2024` directory..

```suggestion
pubDate: 2024-12-12T17:14:23.894+01:00
```

File: src/content/blog/2023/hello-nf-core-world.mdx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+-   **Community News**: Insights into our collective advancements and achievements.
+-   **Technical Insights**: Explanations of decisions and updates from our technical leadership.
+-   **Event Highlights**: Overviews of key takeaways from various nf-core gatherings.
+-   **Tool Updates**: Announcements and insights into the latest nf-core tool releases.
+
+All contributions need to align with the nf-core [code of conduct](https://nf-co.re/code_of_conduct). 
+## Contribution Made Easy
+
+Contributing to this blog is quite straight forward. You just need to add a markdown (or MDX) file to the `src/content/blog` folder.
```
Comment (ewels) [inline src/content/blog/2023/hello-nf-core-world.mdx:40]:
```suggestion
Contributing to this blog is quite straight forward. You just need to add a markdown (or MDX) file to the `src/content/blog/<year>` folder.
```

File: src/content/blog/2023/hello-nf-core-world.mdx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+The file name title of the post separated by dashes. For example, `hello-nf-core-world.mdx`.
+Fill in the frontmatter (the stuff between the three dashes on the top of the markdown files) and start writing your post.
+
+## Follow Along
+
+Similar to the events page, also our blog posts come with an RSS feed. Subscribe to it at [https://nf-co.re/blog/rss.xml](https://nf-co.re/blog/rss.xml) to stay up to date with the latest posts.
```
Comment (ewels) [inline src/content/blog/2023/hello-nf-core-world.mdx:46]: We should create a new slack channel for this to auto-post to as well, and mention it here.

**Review by ewels — CHANGES_REQUESTED** [review]
Nearly there.. I think we're close.

Few minor glitches:

Sidebar overlap with ToC:
CleanShot 2024-01-12 at 16 27 48@2x

Social card doesn't show subtitle:
image

Page layout has space for a sidebar, but nothing there - could be good to have the blog archive as a sidebar?
CleanShot 2024-01-12 at 16 30 17@2x

Feature image is on the listing card, but not on the blog post itself or social card or anything else. Could we also feature it on the detail page, for continuity?
CleanShot 2024-01-12 at 16 31 20@2x

Alternatively to above, could ditch the image and instead have a more traditional "post preview" of each article, with a truncated snippet. Basically like the [Nextflow blog](https://nextflow.io/blog.html), but with the sidebar showing dates (and blog titles? - should be consistent with the sidebar on the detail page).

File: src/pages/blog/[...blogpost].astro [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+const { headings, Content } = await post.render();
+const title = post.data.title;
+const subtitle = post.data.subtitle;
+const md_github_url = 'https://github.com/nf-core/website/blob/main/src/content/blog/' + post.id;
+const headerImage = post.data.headerImage;
+console.log(
+    `/og.png?title=${encodeURIComponent(title)}&subtitle=${encodeURIComponent(
+        subtitle,
+    )}&category=blog&authors=${encodeURIComponent(post.data.authors.join(','))}`,
+);
```
Comment (mashehu) [inline src/pages/blog/[...blogpost].astro:53]:
```suggestion
```

**Review by ewels — APPROVED** [review]
Awesome!

Couple of very minor things, but then LGTM to merge 👍🏻 

Can we have a title / heading on the left hand sidebar? Also, could the index listing page show the blog post titles? (Already does this now on the single post detail page).

CleanShot 2024-01-12 at 19 51 34@2x

File: src/content/blog/2024/hello-nf-core-world.mdx [hunk 7]
```diff
@@ -0,0 +1,49 @@
+---
+title: 'Hello nf-core world'
+subtitle: 'The inital nf-core blog post'
```
Comment (ewels) [inline src/content/blog/2024/hello-nf-core-world.mdx:3]:
```suggestion
subtitle: A new blog for nf-core is launched! Here's what to expect.
```

File: src/pages/blog/index.astro [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+                                            class="card-img-top object-fit-cover"
+                                            alt={post.data.headerImageAlt || ''}
+                                            height={240}
+                                            width={512}
+                                            transition:name="cover-image"
+                                        />
+                                    )}
+                                    <div class="card-body">
+                                        <h2 transition:name="title">{post.data.title}</h2>
+                                        <p class="card-text">
```
Comment (ewels) [inline src/pages/blog/index.astro:65]: Do we want to put the subtitle under here?
```suggestion
                                        <h2 transition:name="title">{post.data.title}</h2>
                                        <p class="lead">{post.data.subtitle}</p>
                                        <p class="card-text small">
```

File: src/pages/blog/index.astro [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+const sections = years.map((year) => {
+    return {
+        section_name: year,
+        section_slug: '#year-' + year,
+        children: blogPosts
+            .filter((post) => new Date(post.data.pubDate).getFullYear() === year)
+            .map((post) => {
+                return {
+                    title: post.data.title,
+                    slug: post.slug,
```
Comment (ewels) [inline src/pages/blog/index.astro:31]:
```suggestion
                    slug: '/blog/' + post.slug,
```

**Review by ewels — APPROVED** [review]
Let's do it!

**Pull request merged**